### PR TITLE
Migrate fra_audit -> assessment_{name}.activity_log

### LINKED ISSUE #1499: Migrate history log
```

/*
 * 'acceptInvitation' = 'invitationAccept'
 * 'addInvitation' = 'invitationAdd'
 * 'createComment' = 'messageCreate'
 * 'createIssue' = 'messageCreate'
 * 'createOdp' = 'originalDataPointCreate'
 * 'deleteComment' = 'messageMarkDeleted'
 * 'deleteOdp' = 'originalDataPointRemove'
 * 'fileRepositoryDelete' = 'assessmentFileDelete'
 * 'fileRepositoryUpload' = 'assessmentFileCreate'
 * 'markAsResolved' = 'topicStatusChange'
 * 'saveDescriptions' = 'descriptionUpdate'
 * 'saveTraditionalTable' = 'nodeValueUpdate'
 * 'updateAssessmentStatus' = 'assessmentStatusUpdate'

 * 'generateFraValues' = 'nodeValueEstimate'
 * 'saveFraValues' = 'originalDataPointUpdate'
 * 'persistGrowingStockValues' = 'nodeValueUpdate'
 */

// Pending implementation

/*
 * 'removeInvitation' = 'invitationRemove',
 * 'removeUser' = 'userRemove',
 * 'updateUser' = 'userUpdate'
 */

// Removed, deprecated

/*
 * 'markAsActual' = not_impleneted: not used anymore
 * 'updateOrInsertDraft' =  not_impleneted: not used anymore
 * 'persistPanEuropeanQuantitativeQuestionnaire' = not_implemented: not part of fra
 */
```

### PR DESCRIPTION
**Pull request by sorja**
resolve  #1499
```/*
 * 'acceptInvitation' = 'invitationAccept'
 * 'addInvitation' = 'invitationAdd'
 * 'createComment' = 'messageCreate'
 * 'createIssue' = 'messageCreate'
 * 'createOdp' = 'originalDataPointCreate'
 * 'deleteComment' = 'messageMarkDeleted'
 * 'deleteOdp' = 'originalDataPointRemove'
 * 'fileRepositoryDelete' = 'assessmentFileDelete'
 * 'fileRepositoryUpload' = 'assessmentFileCreate'
 * 'markAsResolved' = 'topicStatusChange'
 * 'saveDescriptions' = 'descriptionUpdate'
 * 'saveTraditionalTable' = 'nodeValueUpdate'
 * 'updateAssessmentStatus' = 'assessmentStatusUpdate'

 * 'generateFraValues' = 'nodeValueEstimate'
 * 'saveFraValues' = 'originalDataPointUpdate'
 * 'persistGrowingStockValues' = 'nodeValueUpdate'
 */

// Pending implementation

/*
 * 'removeInvitation' = 'invitationRemove',
 * 'removeUser' = 'userRemove',
 * 'updateUser' = 'userUpdate'
 */

// Removed, deprecated

/*
 * 'markAsActual' = not_impleneted: not used anymore
 * 'updateOrInsertDraft' =  not_impleneted: not used anymore
 * 'persistPanEuropeanQuantitativeQuestionnaire' = not_implemented: not part of fra
 */
```

image

### CODE REVIEW

File: tools/dataMigration/migrateActivityLog.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const cycle = assessment.cycles.find((cycle: Cycle) => cycle.name === '2020')
+
+  const schemaName = DBNames.getAssessmentSchema(assessment.props.name)
+  const query = `
+      insert into ${schemaName}.activity_log (
+          time, message, country_iso, section, target, user_id, cycle_uuid
+      )
+      select
+          time,
+          message,
```
Comment (minotogna) [inline tools/dataMigration/migrateActivityLog.ts:17]: Are you 100% sure the message , and all other columns match ? I suspect they don't 
At least the message should be updates

Comment (sorja) [reply tools/dataMigration/migrateActivityLog.ts:17]: I forgot about this, I must check; maybe @rfontanarosa  can help me match to correct ones as he is the author of the new ones

Comment (rfontanarosa) [reply tools/dataMigration/migrateActivityLog.ts:17]: ok, let's talk about it tomorrow after the daily

File: src/meta/assessment/activityLogs.ts [hunk 2]
```diff
@@ -29,7 +32,7 @@ const getLabelActionKey = (activity: ActivityLog<any>) => {
 const getLabelActionParams = (activity: ActivityLog<any>, i18n: i18n) => {
   const { target } = activity
   let params = {}
-  const { user, role, assessment, status, file } = target
+  const { user, role, assessment, status, file } = target ?? {}
```
Comment (sorja) [inline src/meta/assessment/activityLogs.ts:35]: Fix crash when migrated activitylog has no target

**Review by minotogna — APPROVED** [review]

**Pull request merged**